File: patch_inputtray.js
const fs = require('fs');
const file = './src/features/logging/InputTray.jsx';
let content = fs.readFileSync(file, 'utf8');

const regex = /<div className="flex items-center justify-between pb-1 md:pb-0[^>]*>[\s\S]*?<\/button>\s*<\/div>/;

const replacement = `<div className="flex items-center justify-between pb-1 md:pb-0 md:h-10 md:border-b md:border-slate-200 md:bg-slate-50 md:px-3 md:-mx-1.5 md:-mt-3 md:mb-3 md:shrink-0">
          <h2 className="text-xs font-mono font-bold uppercase text-slate-500 md:text-slate-700 leading-tight">
            <span className="hidden md:inline">Attributes</span>
            <span className="md:hidden">Attribute Parameters</span>
          </h2>
          <button 
            type="button" 
            onClick={onClear}
            className="text-[10px] uppercase font-bold text-slate-400 hover:text-slate-700 tracking-wider flex items-center gap-1 cursor-pointer transition-colors"
          >
            <Trash2 className="size-3 md:hidden" />
            <span>Clear</span>
          </button>
        </div>`;

content = content.replace(regex, replacement);
fs.writeFileSync(file, content);
console.log("Patched InputTray");

File: patch_input_tray.py
import re

with open('./src/features/logging/InputTray.jsx', 'r') as f:
    content = f.read()

# add import
content = content.replace(
    "import Input from '../../components/ui/Input';",
    "import Input from '../../components/ui/Input';\nimport SearchableSelect from '../../components/ui/SearchableSelect';"
)

# Replace vendorName select
vendor_regex = re.compile(r'<div className="relative">\s*<select\s*value=\{formData\.vendorName \|\| \'\'\}\s*onChange=\{\(e\) => handleInputChange\(\'vendorName\', e\.target\.value\)\}\s*className=\{selectClass\}\s*>\s*<option value="">Select Vendor...</option>\s*\{VENDOR_NAMES\.map\(v => <option key=\{v\} value=\{v\}>\{v\}</option>\)\}\s*</select>\s*<div className="absolute inset-y-0 right-3 md:right-3\.5 flex items-center pointer-events-none text-slate-400">\s*<ChevronDown className="size-4 md:size-3\.5" />\s*</div>\s*</div>')
new_vendor = '''<SearchableSelect
            value={formData.vendorName || ''}
            onChange={(val) => handleInputChange('vendorName', val)}
            options={VENDOR_NAMES}
            placeholder="Select Vendor..."
            className={selectClass}
          />'''
content = vendor_regex.sub(new_vendor, content)

# Replace structure select
structure_regex = re.compile(r'<div className="relative">\s*<select\s*value=\{formData\.structure \|\| \'\'\}\s*onChange=\{\(e\) => handleInputChange\(\'structure\', e\.target\.value\)\}\s*className=\{selectClass\}\s*>\s*<option value="">Select Structure...</option>\s*\{STRUCTURES\.map\(s => <option key=\{s\} value=\{s\}>\{s\}</option>\)\}\s*</select>\s*<div className="absolute inset-y-0 right-3 md:right-3\.5 flex items-center pointer-events-none text-slate-400">\s*<ChevronDown className="size-4 md:size-3\.5" />\s*</div>\s*</div>')
new_structure = '''<SearchableSelect
            value={formData.structure || ''}
            onChange={(val) => handleInputChange('structure', val)}
            options={STRUCTURES}
            placeholder="Select Structure..."
            className={selectClass}
          />'''
content = structure_regex.sub(new_structure, content)

with open('./src/features/logging/InputTray.jsx', 'w') as f:
    f.write(content)

